sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonValue.java

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.util.serializer.JsonValue;
+import com.google.gson.JsonPrimitive;
+
+import java.util.Objects;
+
+/**
+ * GSON specific implementation of {@link JsonValue}.
+ */
+public final class GsonJsonValue implements JsonValue {
```

Comment:
wondering which one is a more descriptive name for this "core" contract -  `JsonValue` or `JsonPrimitive`

Reply:
Will change to `JsonPrimitive`